conan/tools/cmake/toolchain/blocks.py

```diff
@@ -781,12 +824,17 @@ def context(self):
         cmake_sysroot = self._conanfile.conf.get("tools.build:sysroot")
         cmake_sysroot = cmake_sysroot.replace("\\", "/") if cmake_sysroot is not None else None
 
+        result = self._get_winsdk_version(system_version, generator)
```

Comment:
@memsharded 
Forgive me if I'm wrong: should this be `self._get_winsdk_version(system_version, generator_platfor)` instead?
I have following conf in profile: 
```
tools.cmake.cmaketoolchain:generator=Visual Studio 17 2022
tools.cmake.cmaketoolchain:system_version=10.0.22000
```
and now I get `set(CMAKE_GENERATOR_PLATFORM "Visual Studio 17 2022,version=10.0.22000" CACHE STRING "" FORCE)` in `conan_toolchain.cmake`, which seems incorrect.

Reply:
Hi @mocabe 

Thanks very much for reporting this. Lets have a look.
As a general hint, it is better to create a new ticket (issue) that links to this, that comments in closed PRs, because these might pass unnoticed. (I'll create a ticket for this)